applicationset/generators/pull_request.go

```diff
@@ -140,6 +140,24 @@ func (g *PullRequestGenerator) selectServiceProvider(ctx context.Context, genera
 			return pullrequest.NewBitbucketServiceNoAuth(ctx, providerConfig.API, providerConfig.Project, providerConfig.Repo)
 		}
 	}
+	if generatorConfig.Bitbucket != nil {
+		providerConfig := generatorConfig.Bitbucket
+		if providerConfig.BearerToken != nil {
+			appToken, err := g.getSecretRef(ctx, providerConfig.BearerToken.TokenRef, applicationSetInfo.Namespace)
+			if err != nil {
+				return nil, fmt.Errorf("error fetching Secret Bearer token: %v", err)
+			}
+			return pullrequest.NewBitbucketCloudServiceBearerToken(providerConfig.API, providerConfig.Owner, providerConfig.Repo, appToken)
+		} else if providerConfig.BasicAuth != nil {
+			password, err := g.getSecretRef(ctx, providerConfig.BasicAuth.PasswordRef, applicationSetInfo.Namespace)
+			if err != nil {
+				return nil, fmt.Errorf("error fetching Secret token: %v", err)
+			}
+			return pullrequest.NewBitbucketCloudServiceBasicAuth(providerConfig.BasicAuth.Username, password, providerConfig.API, providerConfig.Owner, providerConfig.Repo)
+		} else {
+			return pullrequest.NewBitbucketCloudServiceNoAuth(providerConfig.API, providerConfig.Owner, providerConfig.Repo)
+		}
+	}
```

Comment:
We can re-write this code as below:

```suggestion
	if generatorConfig.Bitbucket != nil {
		providerConfig := generatorConfig.Bitbucket
		if providerConfig.BasicAuth != nil {
			password, err := g.getSecretRef(ctx, providerConfig.BasicAuth.PasswordRef, applicationSetInfo.Namespace)
			if err != nil {
				return nil, fmt.Errorf("error fetching Secret token: %v", err)
			}
			return pullrequest.NewBitbucketCloudServiceBasicAuth(providerConfig.BasicAuth.Username, password, providerConfig.API, providerConfig.Owner, providerConfig.Repo)
		} else {
			appToken := ""
			// As we do not have a method for NoAuth, we use the Bearer Token as default auth
			if providerConfig.BearerToken != nil {
				appToken, err = g.getSecretRef(ctx, providerConfig.BearerToken.TokenRef, applicationSetInfo.Namespace)
				if err != nil {
					return nil, fmt.Errorf("error fetching Secret Bearer token: %v", err)
				}
			}
			return pullrequest.NewBitbucketCloudServiceBearerToken(providerConfig.API, providerConfig.Owner, providerConfig.Repo, appToken)
		}
	}
```
Let me know what you think.

Reply:
tbh I kinda like the clarity of the extra else-if. I'll leave it as-is, we can always look at a refactor later.